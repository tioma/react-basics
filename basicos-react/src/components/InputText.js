import React, { Component } from 'react';

import PrintValues from './PrintValues';
import CustomInput from './CustomInput';

class InputText extends Component {
  constructor() {
    super();

    this.state = {};

    this.formControls = [{
      name: 'value',
      value: this.state.value,
    }, {
      name: 'value2',
      value: this.state.value2,
    }];

    this.handleChange = this.handleChange.bind(this);
    this.deleteFirstInput = this.deleteInput.bind(this, 'value');
    this.deleteSecondInput = this.deleteInput.bind(this, 'value2');
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  deleteInput(valor) {
    this.setState({ [valor]: '' });
  }

  render() {
    const {
      value3, value4,
    } = this.state;
    return (
      <form>
        <div className="columns">
          <div className="column is-6">
            { this.formControls.map((formControl) => (
              <CustomInput
                key={formControl.name}
                name={formControl.name}
                value={formControl.value}
                handleChange={this.handleChange}
              />
            ))}
          </div>
          <div className="column is-6">
            <CustomInput value={value3} handleChange={this.handleChange} />
            <CustomInput value={value4} handleChange={this.handleChange} />
          </div>
        </div>
        <PrintValues {...this.state} />
        <div className="column">
          <div className="column is-4">
            <div className="field is-grouped">
              <p className="control">
                <button className="button is-primary" type="button" onClick={this.deleteFirstInput}>Borrar input</button>
              </p>
              <p className="control">
                <button className="button is-secondary" type="button" onClick={this.deleteSecondInput}>Borrar input 2</button>
              </p>
            </div>
          </div>
        </div>
      </form>
    );
  }
}

export default InputText;
