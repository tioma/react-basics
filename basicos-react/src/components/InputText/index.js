import React, { Component } from 'react';

import PrintValues from '../PrintValues';
import CustomInput from '../CustomInput/CustomInput';

import CountryContext from '../../Context';

class InputText extends Component {
  constructor(props, context) {
    console.log(context);
    console.log(props);
    super(props, context);

    this.state = {};

    this.formControls = [{
      name: 'value',
      value: this.state.value,
    }, {
      name: 'value2',
      value: this.state.value2,
    }];

    this.handleChange = this.handleChange.bind(this);
    this.deleteFirstInput = this.deleteInput.bind(this, 'value');
    this.deleteSecondInput = this.deleteInput.bind(this, 'value2');
    this.printMessage = this.printMessage.bind(this);
  }

  componentDidMount() {
    setTimeout(() => { this.props.setCountry('cualquier cosa'); }, 6000);
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  deleteInput(valor) {
    this.setState({ [valor]: '' });
  }

  printMessage() {
    const { languaje } = this.context;
    alert(languaje);
  }

  render() {
    const {
      value3, value4,
    } = this.state;
    const { country, languaje } = this.props;
    return (
      <form>
        <div className="columns">
          <div className="column is-6">
            { this.formControls.map((formControl) => (
              <CustomInput
                key={formControl.name}
                name={formControl.name}
                value={formControl.value}
                handleChange={this.handleChange}
              />
            ))}
          </div>
          <div className="column is-6">
            <CustomInput value={value3} handleChange={this.handleChange} />
            <CustomInput value={value4} handleChange={this.handleChange} />
          </div>
        </div>
        <h1>{`Estamos en ${country}`}</h1>
        <h2>{`El lenguaje es ${languaje}`}</h2>
        <PrintValues {...this.state} printMessage={this.printMessage} />
        <div className="column">
          <div className="column is-4">
            <div className="field is-grouped">
              <p className="control">
                <button className="button is-primary" type="button" onClick={this.deleteFirstInput}>Borrar input</button>
              </p>
              <p className="control">
                <button className="button is-secondary" type="button" onClick={this.deleteSecondInput}>Borrar input 2</button>
              </p>
            </div>
          </div>
        </div>
      </form>
    );
  }
}

InputText.contextType = CountryContext;

export default InputText;
