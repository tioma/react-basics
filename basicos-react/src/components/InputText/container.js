import React from 'react';
import { connect } from 'react-redux';

import { actions as userInputActions } from '../../ducks/UserInput';
import { actions as contextActions } from '../../ducks/Context';

import InputText from '.';

const {
  setValue, setValue2, setValue3, setValue4,
} = userInputActions;
const { setCountry } = contextActions;

const InputTextContainer = (props) => (
  <InputText
    value={props.value}
    value2={props.value2}
    value3={props.value3}
    value4={props.value4}
    country={props.country}
    languaje={props.languaje}
    setValue={props.setValue}
    setValue2={props.setValue2}
    setValue3={props.setValue3}
    setValue4={props.setValue4}
    setCountry={props.setCountry}
  />
);

const mapStateToProps = (state) => {
  console.log(state);

  return ({
    value: state.userInput.value,
    value2: state.userInput.value2,
    value3: state.userInput.value3,
    value4: state.userInput.value4,
    country: state.context.country,
    languaje: state.context.languaje,
  });
};

export default connect(
  mapStateToProps,
  {
    setValue,
    setValue2,
    setValue3,
    setValue4,
    setCountry,
  },
)(InputTextContainer);
