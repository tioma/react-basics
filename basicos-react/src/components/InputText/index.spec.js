import React from 'react';
import { shallow } from 'enzyme';

import InputText from '.';

const setup = (overrideProps) => {
  const props = {
    country: 'AR',
    languaje: 'ES',
    setCountry: () => true,
    ...overrideProps,
  };

  const wrapper = shallow(<InputText {...props} />);

  return {
    props,
    wrapper,
    instance: wrapper.instance(),
    buttonPrimary: wrapper.find('button.button.is-primary'),
  };
};

describe('<InputText />', () => {
  it('shoould match snapshot', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  describe('component methods', () => {
    describe('.deleteFirstInput', () => {
      it('should be called on button primary click', () => {
        const deleteInput = jest.spyOn(InputText.prototype, 'deleteInput');
        const { buttonPrimary } = setup();

        expect(buttonPrimary.length).toBe(1);

        buttonPrimary.simulate('click');

        expect(deleteInput).toBeCalled();
        expect(deleteInput).toBeCalledWith('value');
        deleteInput.mockClear();
      });
    });

    describe('.deleteInput', () => {
      it('should delete state property given as argument', () => {
        const { instance } = setup();

        expect(instance.state).toMatchObject({});

        instance.setState({ value: 'something' });

        expect(instance.state.value).toBe('something');
        instance.deleteInput('value');

        expect(instance.state.value).toBe('');
      });
    });
  });
});
