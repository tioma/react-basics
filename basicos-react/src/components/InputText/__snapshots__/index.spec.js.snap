// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> shoould match snapshot 1`] = `
<form>
  <div
    className="columns"
  >
    <div
      className="column is-6"
    >
      <Wrapper
        handleChange={[Function]}
        key="value"
        name="value"
      />
      <Wrapper
        handleChange={[Function]}
        key="value2"
        name="value2"
      />
    </div>
    <div
      className="column is-6"
    >
      <Wrapper
        handleChange={[Function]}
      />
      <Wrapper
        handleChange={[Function]}
      />
    </div>
  </div>
  <h1>
    Estamos en AR
  </h1>
  <h2>
    El lenguaje es ES
  </h2>
  <Wrapper
    printMessage={[Function]}
  />
  <div
    className="column"
  >
    <div
      className="column is-4"
    >
      <div
        className="field is-grouped"
      >
        <p
          className="control"
        >
          <button
            className="button is-primary"
            onClick={[Function]}
            type="button"
          >
            Borrar input
          </button>
        </p>
        <p
          className="control"
        >
          <button
            className="button is-secondary"
            onClick={[Function]}
            type="button"
          >
            Borrar input 2
          </button>
        </p>
      </div>
    </div>
  </div>
</form>
`;
